### PR TITLE
Add "fport" option to the TinyLoRa library

### PR DESCRIPTION
**Pull request by mattrude**
This is a fairly simple patch to add "fport" option when sending messages via the tinyLoRa library to the "The Things Network"

This change allows to have more then a single device type per application on the things network.

### CODE REVIEW

File: adafruit_tinylora/adafruit_tinylora.py [hunk 1]
```diff
@@ -265,7 +267,7 @@ def send_data(self, data, data_length, frame_counter, timeout=2):
         lora_pkt[5] = 0
         lora_pkt[6] = frame_counter & 0x00FF
         lora_pkt[7] = (frame_counter >> 8) & 0x00FF
-        lora_pkt[8] = 0x01
+        lora_pkt[8] = int('0x{0:02x}'.format(self._fport))
```
Comment (jerryneedell) [inline adafruit_tinylora/adafruit_tinylora.py:270]: I don't understand this statement. Shouldn't this just be 
```
lora_pkt[8] = self._fport  
```
?

File: adafruit_tinylora/adafruit_tinylora.py [hunk 2]
```diff
@@ -130,7 +130,7 @@ class TinyLoRa:
     _BUFFER = bytearray(2)
 
     # pylint: disable=too-many-arguments
-    def __init__(self, spi, cs, irq, rst, ttn_config, channel=None):
+    def __init__(self, spi, cs, irq, rst, ttn_config, channel=None, fport=1):
```
Comment (jerryneedell) [inline adafruit_tinylora/adafruit_tinylora.py:133]: In the Arduino Library the fport is passed as an argument to send_data  not to init ?
I am not familiar with its usage, but just noticed the difference.
Does it matter?

https://github.com/adafruit/TinyLoRa/blob/master/TinyLoRa.cpp#L610

Comment (brentru) [reply adafruit_tinylora/adafruit_tinylora.py:133]: Agree with @jerryneedell on this point. I feel the frame port should be `0x01` by default and optionally set with a payload in the send function.

Specifically, I'd like to see `fport` defined as a kwarg in `send_data` here: (https://github.com/adafruit/Adafruit_CircuitPython_TinyLoRa/blob/master/adafruit_tinylora/adafruit_tinylora.py#L238) and set here (https://github.com/adafruit/Adafruit_CircuitPython_TinyLoRa/blob/master/adafruit_tinylora/adafruit_tinylora.py#L268).

Comment (jerryneedell) [reply adafruit_tinylora/adafruit_tinylora.py:133]: Is there any need for it to be saved as an attribute at all? should it just be passed in via a kwarg fport and sent as 
```
lora_pkt[8] = fport
```
?

Comment (brentru) [reply adafruit_tinylora/adafruit_tinylora.py:133]: Oh, my mistake. We'd want to change `send_packet` at the high-level, and then pass the frame port to `send_data` where it's actually set in the `lora_pkt`

 `def send_packet(self, lora_packet, packet_length, frame_port, timeout):`

**Comment by kattni** [comment]
@mattrude Are you available to follow up on this? Thanks!

**Comment by brentru** [comment]
@mattrude  I am closing this for now. When you are available to follow up on this PR, please ping me here and I'll re-open.

**Pull request closed without merging**